properties.h

```diff
@@ -1,13 +1,25 @@
 #ifndef __properties_H_
 #define __properties_H_
 
-#ifndef EXTERN
-#ifdef __cplusplus
-#define EXTERN extern "C"
+#ifdef LIBPROPERTIES_EXPORT
+#ifdef _WIN32
+#define LIBPROPERTIES_EXTERN __declspec(dllexport)
+#else
+#define LIBPROPERTIES_EXTERN __attribute__((visibility("default")))
+#endif
+#elif !defined(LIBPROPERTIES_STATIC)
+#ifdef _WIN32
+#define LIBPROPERTIES_EXTERN __declspec(dllimport)
 #else
-#define EXTERN
+#define LIBPROPERTIES_EXTERN
+#endif
+#else
+#define LIBPROPERTIES_EXTERN
+#endif
+
+#ifdef __cplusplus
+#define EXTERN extern "C" {
```

Comment:
```suggestion
extern "C" {
```

Reply:
@libbylg
Don't forget to fix this one.
Maybe also add a test, using your library by a c++ source